airbyte-workers/src/main/java/io/airbyte/workers/storage/StateStore.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.storage;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.State;
+import io.airbyte.config.storage.CloudStorageConfigs.GcsConfig;
+import io.airbyte.config.storage.CloudStorageConfigs.MinioConfig;
+import io.airbyte.config.storage.CloudStorageConfigs.S3Config;
+import io.airbyte.config.storage.DefaultGcsClientFactory;
+import io.airbyte.config.storage.DefaultS3ClientFactory;
+import io.airbyte.config.storage.MinioS3ClientFactory;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Leverages the CloudDocumentStore. The root directory is state and then each activityRunId is the
+ * key.
+ */
+public class StateStore {
+
+  private static final Path STATE_ROOT = Path.of("state");
+
+  private final CloudDocumentStoreClient documentStoreClient;
+
+  public StateStore s3(final S3Config config) {
+    return new StateStore(new S3CloudDocumentStoreClient(
+        new DefaultS3ClientFactory(config).get(),
+        config.getBucketName(),
+        STATE_ROOT));
+  }
+
+  public StateStore minio(final MinioConfig config) {
+    return new StateStore(new S3CloudDocumentStoreClient(
+        new MinioS3ClientFactory(config).get(),
+        config.getBucketName(),
+        STATE_ROOT));
+  }
+
+  public StateStore gcs(final GcsConfig config) {
+    return new StateStore(new GcsCloudDocumentStoreClient(
+        new DefaultGcsClientFactory(config).get(),
+        config.getBucketName(),
+        STATE_ROOT));
+  }
```

Comment:
these can move to where ever it convenient. i kinda figure when we actually plug this into the worker it will become obvious if this is a good place for them, or if we ant to move them. 

@davinchia this is where we are leveraging those factories that you were mentioning as redundant in the previous PR. I think they could probably just be static helpers instead of class that gets instantiated. They let us reuse the same configuration object for configuring cloud storage for here as we do for logs, so we don't have to duplicate the logic!

Reply:
I see!

The static pattern makes sense to me, since there isn't state to manage here.

I also have slight preference for a single class that handles all storage client creation instead of 3 separate small factories.

Reply:
kk. i'm going to do it separately. but agreed.

Reply:
moved the static factories as jared suggested.